### ISSUE TITLE
Silence two build warnings on systems where ivsize > ptrsize.

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12559]
Migrated from [rt.perl.org#115710](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=115710) (status was 'resolved')

Searchable as RT115710$

**Comment by p5pRT** [comment]
### From @ikegami

#### Created by @ikegami




Attached patch silences two build warnings on systems where ivsize >
ptrsize\.

They are safe to ignore\, a side\-effect of a function with a polymorphic
interface\.

cv = find\_runcv\_where\(FIND\_RUNCV\_level\_eq\, iv\,        NULL\);
cv = find\_runcv\_where\(FIND\_RUNCV\_padid\_eq\, PTR2IV\(p\)\, NULL\);  // p is a
PADNAMELIST\*

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=low

Site configuration information for perl 5.14.2:

Configured by rurban at Thu Jul 12 13:58:56 CDT 2012.

Summary of my perl5 (revision 5 version 14 subversion 2) configuration:

  Platform:
    osname=cygwin, osvers=1.7.15(0.26053),
archname=cygwin-thread-multi-64int
    uname='cygwin_nt-5.1 winxp 1.7.15(0.26053) 2012-05-09 10:25 i686 cygwin
'
    config_args='-de -Dlibperl=cygperl5_14.dll -Dcc=gcc-4 -Dld=g++-4
-Darchname=i686-cygwin-threads-64int -Dmksymlinks -Dusethreads -Accflags=-g'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc-4', ccflags ='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__ -g
-fno-strict-aliasing -pipe -fstack-protector',
    optimize='-O3',
    cppflags='-DPERL_USE_SAFE_PUTENV -U__STRICT_ANSI__ -g
-fno-strict-aliasing -pipe -fstack-protector'
    ccversion='', gccversion='4.5.3', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='g++-4', ldflags =' -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--enable-auto-image-base -fstack-protector
-L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib
    libs=-lgdbm -ldb -ldl -lcrypt -lgdbm_compat
    perllibs=-ldl -lcrypt
    libc=/usr/lib/libc.a, so=dll, useshrplib=true, libperl=cygperl5_14.dll
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' --shared  -Wl,--enable-auto-import
-Wl,--export-all-symbols -Wl,--enable-auto-image-base -L/usr/local/lib
-fstack-protector'

Locally applied patches:
    Bug#55162 File::Spec::case_tolerant performance
    CYG07 $vendorarch/auto/.rebase
    CYG15 static Win32CORE
    CYG17 cyg-1.7 paths-utf8
    0c612ce82 Fix building static extensions on cygwin, -UUSEIMPORTLIB
    1bac5ecc1 Fix 64-bit threading sv.c: S_anonymise_cv_maybe
    Cygwin::sync_winenv added


@INC for perl 5.14.2:
    /usr/lib/perl5/site_perl/5.14/i686-cygwin-threads-64int
    /usr/lib/perl5/site_perl/5.14
    /usr/lib/perl5/vendor_perl/5.14/i686-cygwin-threads-64int
    /usr/lib/perl5/vendor_perl/5.14
    /usr/lib/perl5/5.14/i686-cygwin-threads-64int
    /usr/lib/perl5/5.14
    /usr/lib/perl5/site_perl/5.10
    /usr/lib/perl5/vendor_perl/5.10
    /usr/lib/perl5/site_perl/5.8
    .


Environment for perl 5.14.2:
    HOME=/home/ikegami
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)

PATH=/home/ikegami/usr/perlbrew/bin:.:/home/ikegami/bin:/usr/local/bin:/usr/bin:/cygdrive/c/bin:/cygdrive/c/progs/perl5161-ap1601/site/bin:/cygdrive/c/progs/perl5161-ap1601/bin:/cygdrive/c/Windows/system32:/cygdrive/c/Windows
    PERLBREW_BASHRC_VERSION=0.42
    PERLBREW_HOME=/home/ikegami/.perlbrew
    PERLBREW_PATH=/home/ikegami/usr/perlbrew/bin
    PERLBREW_ROOT=/home/ikegami/usr/perlbrew
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @ikegami

<details><summary>0001-Silence-two-build-warnings-when-ivsize-ptrsize.patch</summary>

```diff
From d49cc213f4a65891c47ac9d2525f0e3d4d0d427c Mon Sep 17 00:00:00 2001
From: Eric Brine <ikegami@adaelis.com>
Date: Tue, 13 Nov 2012 05:38:30 -0500
Subject: [PATCH] Silence two build warnings when ivsize > ptrsize.

They are safe to ignore, a side-effect of a function with a polymorphic interface.

cv = find_runcv_where(FIND_RUNCV_level_eq, iv,        NULL);
cv = find_runcv_where(FIND_RUNCV_padid_eq, PTR2IV(p), NULL);  // p is a PADNAMELIST*
---
 pad.c    |    2 +-
 pp_ctl.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/pad.c b/pad.c
index 2292aaf..cc34ade 100644
--- a/pad.c
+++ b/pad.c
@@ -1999,7 +1999,7 @@ S_cv_clone_pad(pTHX_ CV *proto, CV *cv, CV *outside)
 	    || PadlistNAMES(CvPADLIST(outside))
 		 != protopadlist->xpadl_outid) {
 	    outside = find_runcv_where(
-		FIND_RUNCV_padid_eq, (IV)protopadlist->xpadl_outid, NULL
+		FIND_RUNCV_padid_eq, PTR2IV(protopadlist->xpadl_outid), NULL
 	    );
 	    /* outside could be null */
 	}
diff --git a/pp_ctl.c b/pp_ctl.c
index 1f4afc0..5374e0f 100644
--- a/pp_ctl.c
+++ b/pp_ctl.c
@@ -3263,7 +3263,7 @@ Perl_find_runcv_where(pTHX_ U8 cond, IV arg, U32 *db_seqp)
 		switch (cond) {
 		case FIND_RUNCV_padid_eq:
 		    if (!CvPADLIST(cv)
-		     || PadlistNAMES(CvPADLIST(cv)) != (PADNAMELIST *)arg)
+		     || PadlistNAMES(CvPADLIST(cv)) != INT2PTR(PADNAMELIST *, arg))
 			continue;
 		    return cv;
 		case FIND_RUNCV_level_eq:
-- 
1.7.9


```
</details>

**Comment by p5pRT** [comment]
### From @khwilliamson

On 11/13/2012 03&#8203;:42 AM\, Eric Brine \(via RT\) wrote&#8203;:
> \# New Ticket Created by  "Eric Brine"
> \# Please include the string&#8203;:  \[perl \#115710\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=115710 >
>
>
> This is a bug report for perl from ikegami@&#8203;adaelis\.com\,
> generated with the help of perlbug 1\.39 running under perl 5\.14\.2\.
>
>
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
>
>
> Attached patch silences two build warnings on systems where ivsize >
> ptrsize\.
>
> They are safe to ignore\, a side\-effect of a function with a polymorphic
> interface\.
>
> cv = find\_runcv\_where\(FIND\_RUNCV\_level\_eq\, iv\,        NULL\);
> cv = find\_runcv\_where\(FIND\_RUNCV\_padid\_eq\, PTR2IV\(p\)\, NULL\);  // p is a
> PADNAMELIST\*
>
>
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
>

Thanks\, applied as
a56015b9dc044df9d25d61522b810c3a5043b5df

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @khwilliamson

Resolved with a56015b9dc044df9d25d61522b810c3a5043b5df
\-\- 
Karl Williamson

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'open' to 'resolved'
